analysis/paper/paper.Rmd

```diff
@@ -325,207 +328,142 @@ For our single cell purposes, one of the primary modifications we make is the in
-    )
+lapply(
+  seq_along(assignment.list),
+  function(x) {
+    for (i in 1:length(assignment.list[[x]])) {
+      contrastConstructor(
+        clusterIDs = assignment.list[[x]][[i]],
+        clusterIDs.2 = setdiff(assignment.list[[x]], assignment.list[[x]][[i]]),
+        design = edgeRobj$design,
+        makeContrast = T,
+        name = paste(names(assignment.list)[[x]], "subset", assignment.list[[x]][[i]], sep = "_")
+      )
+    }
   }
+)
 
-  grep(ls(), pattern = "DC[0-9]", value = T)
-  DC.celltype.contrasts <- do.call(cbind, lapply(grep(ls(), pattern = "DC[0-9]", value = T), get))
-  colnames(DC.celltype.contrasts) <- grep(ls(), pattern = "DC[0-9]", value = T)
-}
+subset.cluster.contrasts <- do.call(cbind, lapply(grep(ls(), pattern = "_subset_", value = T), get))
+subset.cluster.contrasts <- as.data.frame(subset.cluster.contrasts)
+colnames(subset.cluster.contrasts) <- grep(ls(), pattern = "_subset_", value = T)
 ```
 
 ## Part 6.3: Conducted differential gene expression testing between major cell groups and between groups
 
 ```{r Conduct differential gene expression testing for major cell groups and subsets}
-major.glmQLFTest <- list()
-for (i in 1:length(colnames(major.contrasts))) {
-  major.glmQLFTest[[i]] <- glmQLFTest(fit, contrast = major.contrasts[, i]) %>%
+major_dge_list <- list()
+for (i in 1:length(colnames(major.group.contrasts))) {
+  major_dge_list[[i]] <- glmQLFTest(edgeRobj$fit, contrast = major.group.contrasts[, i]) %>%
     topTags(n = NULL)
 }
 
-Myeloid.glmQLFTest <- list()
-for (i in 1:length(colnames(M.celltype.contrasts))) {
-  Myeloid.glmQLFTest[[i]] <- glmQLFTest(fit, contrast = M.celltype.contrasts[, i]) %>%
+subset_dge_list <- list()
+for (i in 1:length(colnames(subset.cluster.contrasts))) {
+  subset_dge_list[[i]] <- glmQLFTest(edgeRobj$fit, contrast = subset.cluster.contrasts[, i]) %>%
     topTags(n = NULL)
 }
+```
 
-Mo.glmQLFTest <- list()
-for (i in 1:length(colnames(Mo.celltype.contrasts))) {
-  Mo.glmQLFTest[[i]] <- glmQLFTest(fit, contrast = Mo.celltype.contrasts[, i]) %>%
-    topTags(n = NULL)
-}
+## Part 6.4: Append percent.expression feature data and putative surface protein classification to differential gene expression tables
 
-DC.glmQLFTest <- list()
-for (i in 1:length(colnames(DC.celltype.contrasts))) {
-  DC.glmQLFTest[[i]] <- glmQLFTest(fit, contrast = DC.celltype.contrasts[, i]) %>%
-    topTags(n = NULL)
-}
+```{r Append percent.expression feature data and putative surface protein classification to differential gene expression tables}
+major_dge_list <- add.percent.expression(
+  dge_list = major_dge_list, object = ePBMC.integrated.obj,
+  ident = "MajorGroup"
+)
 
-for (i in 1:length(colnames(B.celltype.contrasts))) {
-  B.glmQLFTest[[i]] <- glmQLFTest(fit, contrast = B.celltype.contrasts[, i]) %>%
-    topTags(n = NULL)
-}
+major.groups <- unique(ePBMC.integrated.obj$MajorGroup)
+
+b.dge.list <- add.percent.expression(
+  dge_list = mono.dge.list,
```

Comment:
Is this a typo? Object is "b dge list" but you are specifying 'mono dge list" in the function

Reply:
Fixed. This should be referring the 'subset_dge_list'. This list will contains the all of the topTag tables for all of the subset (within major group) DGE tests.

Upon specification of the group.ident and subset, the cluster IDs will be pulled directly from the Seurat object and the percent expressed values will be calculated appropriately.